tests/test_conftest.py

```
@@ -48,6 +50,58 @@ def test_template_drops_sample_hooks_but_keeps_hooks_dir_and_exclude(project):
+    assertion message. The trace-recorded-the-commit assertion is the anti-vacuity
+    guard: `spawned` reads empty both when no child ran and when the trace parsed
+    into nothing we recognize, so a trace2 event or field rename would otherwise
+    leave this row green for having observed nothing. It does not guard an absent
+    trace — a git without trace2, or a mistyped env var, writes no file at all and
+    the read below raises `FileNotFoundError`, which is loud on its own."""
+    repo = project.project
+    (repo / "src.txt").write_text("changed\n")
+    subprocess.run(["git", "-C", str(repo), "add", "-A"], check=True, capture_output=True)
+
+    trace = tmp_path / "trace2.json"
+    subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)